src/App.tsx

```diff
@@ -1,10 +1,38 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { ErrorMessage } from './components/ErrorMessage';
+import { Error } from './types/Error';
+import { filterTodos } from './utils/filterTodos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterStatus, setFilterStatus] = useState<Filter>(Filter.All);
+  const [error, setError] = useState<Error | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(resolve => setTodos(resolve))
+      .catch(() => setError(Error.UnableToLoadAll));
+  }, []);
+
+  const filteredTodos = useCallback(() => {
```

Comment:
```suggestion
  const filteredTodos = useMemo(() => {
```
and do not call it below, just use array